alphadia/workflow/peptidecentric/optimization_handler.py

```diff
@@ -60,7 +58,39 @@ def __init__(
         self.dia_data = dia_data
         self.figure_path = figure_path
 
-        self.optlock: OptimizationLock | None = None
+        self.optlock: OptimizationLock = OptimizationLock(
+            self.spectral_library, self.config
+        )
+
+    def _init_optimiser(
+        self,
+        clazz: type[AutomaticOptimizer | TargetedOptimizer],
+        initial_parameter: float,
+        target_parameter: float | None = None,
+    ):
+        """Returns an instance of the specified optimizer class with the given initial and target parameters."""
+        if issubclass(clazz, TargetedOptimizer):
+            return clazz(
```

Comment:
"Clazz" can refer to a Java type used in programming or a German a cappella choir. In Java, "clazz" is a common variable name used to represent a class object, often when the keyword "class" is already in use. "CLAZZ" (with all caps) is also the name of a German choir specializing in a cappella arrangements of pop and jazz music.

Reply:
Learned something today